gridded_etl_tools/utils/extractor.py

```diff
@@ -418,3 +397,35 @@ def batch_requests(self, pattern: str = ".*") -> list[pathlib.PurePosixPath]:
             List of paths to files in the current working directory matching the pattern
         """
         return list(self.find(pattern))
+
+
+def get_hrefs(session: requests.Session, url: str, filters: tuple[str]) -> tuple[str]:
+    """
+    Get a list of links from a given URL.
+
+    Parameters
+    ----------
+    session : requests.Session
+        An existing session object to make requests.
+    url : str
+        A URL to scrape links from.
+    filters : tuple[str]
```

Comment:
I still think filtering against just the beginning of the url string is a bit too specific for a library function, especially considering that BeautifulSoup provides a general filter that could alternatively be passed.

We do in many cases use other filters like "get all *.nc4" or "get everything except *.txt" etc.

In the case of excluding https/mailto, the filter would be a custom function (which can be passed as a lambda since it's a short function)

```
soup.find_all("a", lambda href: not href.startswith("https://") and not href.startswith("mailto://"))
```

So what I'm proposing is that the filter argument to `extractor.get_hrefs` would accept a function rather that a list of strings, so that the following would be the call to get all links that don't start with "https://" or "mailto://".

```
extractor.get_hrefs(session, url, lambda href: not href.startswith("https://") and not href.startswith("mailto://"))
```